app/controllers/api/v1/jwt_api/protocol_subscriptions_controller.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module JwtApi
+      class ProtocolSubscriptionsController < JwtApiController
+        def mine
+          render json: current_user.protocol_subscriptions, each_serializer: Api::ProtocolSubscriptionSerializer
```

Comment:
Shouldn't this be `current_auth_user.person`? Because if we use current_user, doesn't that open up the possibilities of accepting the other methods of authentication?

Or better still: ideally, we move the definition for current_user to the respective JwtApiController, JwtApiAndCookieController, BasicAuthController, and then let those definitions of current_user only exist of the allowed authentication methods in those respective classes.

It's not really a security concern, because to get into e.g., the basic auth controller you need valid basic auth, but if you also supply e.g., an Authentication header with a Bearer token, AND we happen to use current_user somewhere in a method of a basic auth controller (or deriving class), then a call to a basic_auth controller method will create an auth user and its person, just because we happened to pass a header along, which is probably something we don't want.

Reply:
hmm yeah I think this really depends on what we want. I like the fact that we have different ways of authenticating the user (that was originally how I created the api). And in that case, it is nice if there is just one function that we can use. What I did now is use current_user everywhere in the subclasses. We can split it up if we want later.

Reply:
Ok

Reply:
It's just that in most other classes I saw `current_auth_user.person` rather than `current_user`.